File: DSA/Arshdeep-Kaur/Day 40/Day40_Question1.cpp
class Solution {
public:
    int minSwap(int arr[], int n, int k) {
        int nums = 0;
        for(int i=0; i<n; i++) {
            if (arr[i] <= k) {
                nums++;
            }
        }
        
        int curr_wndw =  0;
        for(int i=0; i<nums; i++) {
            if(arr[i] <= k) {
                curr_wndw++;
            }
        }
        
        int ans = nums - curr_wndw;
        for(int i=nums; i<n; i++) {
            if (arr[i] <= k) {
                curr_wndw++;
            }
            if (arr[i-nums] <= k) {
                curr_wndw--;
            }
            
            ans = min(ans, nums - curr_wndw);
        }
        
        return ans;
    }
};

File: DSA/Manav-Rajput/Day40_question1.java

class Complete {
    public static int minSwap(int[] arr, int n, int k) {
        int countLessOrEqualK = 0;
        for (int i = 0; i < n; i++) {
            if (arr[i] <= k) {
                countLessOrEqualK++;
            }
        }
        int unwanted = 0;
        for (int i = 0; i < countLessOrEqualK; i++) {
            if (arr[i] > k) {
                unwanted++;
            }
        }
        int minSwaps = unwanted;
        for (int i = 0, j = countLessOrEqualK; j < n; i++, j++) {
          
            if (arr[i] > k) {
                unwanted--;
            }
           
            if (arr[j] > k) {
                unwanted++;
            }
            minSwaps = Math.min(minSwaps, unwanted);
        }
        
        return minSwaps;
    }
}
